src/commands.cpp

```diff
@@ -1161,7 +1161,7 @@ void CommandLogout::procresult()
 {
     error e = (error)client->json.getint();
     MegaApp *app = client->app;
-    if(!e)
+    if(!e || e == API_ESID)
```

Comment:
I think that API_ESID can't be received as a command-level response, only as a request-level response.